### DISCUSSION TITLE
How to contribute? - I got error like 'env required' but I have set up env

### DISCUSSION THREAD

**Discussion by nicitaacom** [supabase/supabase#19306] [Contribute to Supabase]
## What I'm trying to do
I want to contribute to supabase

## Error that I have
![image](https://github.com/supabase/supabase/assets/39565703/f75507c1-148a-45ff-8454-f57ee2f2fc68)

## How I got this error
1. I forked supabase repository
2. `npm i`
3. `npm run dev`
4. got this error

Also I tried `npm run dev:www` but in that case I got `document is not defined` error
So how to fix this?
I mean provide some step by step guide or YouTube video
I used docs DEVELOPERS.md and

**Comment by dcourv** [comment]
I would also like to start contributing but ran into the same error

**Comment by botxboom** [comment]
Hi @nicitaacom and @dcourv , have you run the docker containers ?
you need to setup env variables. checkout the link below and follow the steps for running docker for supabase studio.
https://github.com/supabase/supabase/blob/master/DEVELOPERS.md#get-started

**Comment by dcourv** [comment]
Hey @botxboom, I was able to access studio running through docker at localhost:8000 after entering basic authentication, but was not able to run any of the sites using the local development server as listed in the docs (https://github.com/supabase/supabase/blob/master/DEVELOPERS.md#install-dependencies-1)

**Reply by nicitaacom** [reply]
I'm getting error when I run `docker compose up`
```
docker compose up
docker: 'compose' is not a docker command.
See 'docker --help'
```

I tried to install
`curl -sSL https://get.docker.com/ | sudo sh`
also I tried to install `docker-compose` and run this command as `docker-compose up`

**Reply by botxboom** [reply]
> I'm getting error when I run `docker compose up`
> ```
> docker compose up
> docker: 'compose' is not a docker command.
> See 'docker --help'
> ```
> 
> I tried to install
> `curl -sSL https://get.docker.com/ | sudo sh`
> also I tried to install `docker-compose` and run this command as `docker-compose up`

which OS you are using ?
make sure you have docker deamon (GUI) up and running.

This might help.

https://github.com/docker/compose/issues/8630#issuecomment-1141930536

**Reply by nicitaacom** [reply]
> which OS you are using ?

kali linux

> make sure you have docker deamon (GUI) up and running

How can I make sure in that?

> This might help.
>https://github.com/docker/compose/issues/8630#issuecomment-1141930536

I entered that 2 commands and then in supabase directory I ran `docker compose up`
And got this error
```
docker: 'compose' is not a docker command.
See 'docker --help'
```

**Reply by botxboom** [reply]
> > which OS you are using ?
> 
> kali linux
> 
> > make sure you have docker deamon (GUI) up and running
> 
> How can I make sure in that?
> 
> > This might help.
> >https://github.com/docker/compose/issues/8630#issuecomment-1141930536
> 
> I entered that 2 commands and then in supabase directory I ran `docker compose up`
> And got this error
> ```
> docker: 'compose' is not a docker command.
> See 'docker --help'
> ```

try:

`sudo apt-get install docker-compose`

check if it is installed:

`docker-compose version`

then:

`docker-compose up -d`
or 
`docker-compose up`

**Reply by botxboom** [reply]
Hi @nicitaacom , I have tried above stesps in ubuntu. 
should work for you as well. Let me know if you still face issue.

**Reply by nicitaacom** [reply]
I'm still face issue

```
docker-compose version
docker-compose version 1.29.2, build unknown
docker-py version: 5.0.3
CPython version: 3.11.5
OpenSSL version: OpenSSL 3.0.11 19 Sep 2023
```

![image](https://github.com/supabase/supabase/assets/39565703/fa58647f-82eb-4cd4-aa2c-399327e614de)

**Comment by nicitaacom** [comment]
@botxboom 
Thank you
Commands from this site helped - https://www.digitalocean.com/community/tutorials/how-to-install-and-use-docker-compose-on-ubuntu-22-04

```bash
mkdir -p ~/.docker/cli-plugins/
curl -SL https://github.com/docker/compose/releases/download/v2.3.3/docker-compose-linux-x86_64 -o ~/.docker/cli-plugins/docker-compose
chmod +x ~/.docker/cli-plugins/docker-compose
docker compose version
```

**Comment by nicitaacom** [comment]
# Fixed

To fix this issue you need to do the followind

### Step 1 - make sure you correctly installed docker

```bash
sudo apt-remove docker
```

```
mkdir -p ~/.docker/cli-plugins/
curl -SL https://github.com/docker/compose/releases/download/v2.3.3/docker-compose-linux-x86_64 -o ~/.docker/cli-plugins/docker-compose
chmod +x ~/.docker/cli-plugins/docker-compose
docker compose version
```

### Step 2 - copy env and run docker
```
cd docker
cp .env.example .env
docker compose up
```

### Step 3 - fix this issue
If you run `npm run dev:www` as me you need to setup env for `www forlder`
For this go to this file `apps/www/.env`
![image](https://github.com/supabase/supabase/assets/39565703/d70d26e5-e572-4260-97c4-49882fad579d)

then paste `NEXT_PUBLIC_SUPABASE_URL` and `NEXT_PUBLIC_SUPABASE_ANON_KEY` from any your project
Or you may create new supabase project and get this `.env` values from there
How to create new project in supabase and setup .env - https://youtu.be/4AL9uqG8Svk?list=LL&t=941
